.github/workflows/ci.yml

```
@@ -18,68 +18,41 @@ jobs:
-          fetch-depth: 0
-      - name: Run pre-commit-conda
-        uses: quantco/pre-commit-conda@v1
-
-  mypy-type-checks:
-    name: Mypy Type Checks
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout branch
-        uses: actions/checkout@v4
-        with:
-          ref: ${{ github.head_ref }}
-          fetch-depth: 0
-      - name: Set up Conda env
-        uses: mamba-org/setup-micromamba@422500192359a097648154e8db4e39bdb6c6eed7
-        with:
-          condarc-file: .github/assets/.condarc
-          environment-file: environment.yml
-          cache-environment: true
-      - name: Install repository
-        run: python -m pip install --no-build-isolation --no-deps --disable-pip-version-check -e .
-      - name: Run mypy
-        run: mypy .
+          environments: default lint
+      - name: pre-commit
+        run: pixi run pre-commit-run --color=always --show-diff-on-failure
 
   unit-tests:
-    name: Unit Tests - ${{ matrix.os == 'ubuntu-latest' && 'Linux' || 'Windows' }} - Python ${{ matrix.python-version }}
+    name: Unit Tests
     timeout-minutes: 30
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
-        python-version: ["3.10", "3.11", "3.12"]
+        os: [ubuntu-latest, macos-latest]
+        env: ["py310", "py311", "py312"]
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
         with:
-          ref: ${{ github.head_ref }}
-          fetch-depth: 0
-      - name: Set up Conda env
-        uses: mamba-org/setup-micromamba@422500192359a097648154e8db4e39bdb6c6eed7
-        with:
-          condarc-file: .github/assets/.condarc
-          environment-file: environment.yml
-          cache-environment: true
-          create-args: >-
-            python=${{ matrix.python-version }}
-            pytest-md
-            pytest-emoji
+          environments: ${{ matrix.env }}
       - name: Install repository
-        run: python -m pip install --no-build-isolation --no-deps --disable-pip-version-check -e .
-      - name: Run unittests
-        uses: quantco/pytest-action@v2
+        run: |
+          pixi run -e ${{ matrix.env }} postinstall
+          pixi run -e ${{ matrix.env }} coverage
+      - name: Generate code coverage report
+        if: matrix.python-version == '3.12'
+        uses: codecov/codecov-action@v3.1.3
         with:
-          report-title: Unit Tests - ${{ matrix.os == 'ubuntu-latest' && 'Linux' || 'Windows' }} - Python ${{ matrix.python-version }}
-          custom-arguments: --cov=metalearners --cov-report=xml --cov-report term-missing --color=yes
+          file: ./coverage.xml
       - name: Upload coverage reports to Codecov
-        if: matrix.python-version == '3.12'
-        uses: codecov/codecov-action@125fc84a9a348dbcf27191600683ec096ec9021c
+        if: ${{ matrix.env }} == "py312"
```

```suggestion
        if: matrix.env == "py312"
```

This should work as well